stl/inc/flat_set

```diff
@@ -153,6 +153,27 @@ public:
     _Base_flat_set(_Tsorted _Tsort, initializer_list<_Kty> _Ilist, const _Alloc& _Al)
         : _Base_flat_set(_Tsort, container_type(_Ilist, _Al)) {}
 
+    _Base_flat_set(const _Base_flat_set&) = default;
+    _Base_flat_set(_Base_flat_set&& _Other)
+        : _Mycont(_STD move(_Other).extract()), _Mycomp(_Other._Mycomp) // intentionally copy comparator, see LWG-2227
+    {}
+
+    _Base_flat_set& operator=(const _Base_flat_set& _Other) {
+        _Clear_guard<_Base_flat_set> _Guard{this};
+        _Mycont        = _Other._Mycont;
+        _Mycomp        = _Other._Mycomp;
+        _Guard._Target = nullptr;
+        return *this;
+    }
+    _Base_flat_set& operator=(_Base_flat_set&& _Other) {
+        _Clear_guard<_Base_flat_set> _Guard{this};
+        _Clear_guard<_Base_flat_set> _Always_clear{_STD addressof(_Other)};

```

Comment:
`_Guard` is needed when:
1. the move assignment fails and cannot provide strong-guarantee on `this` side, or
2. _Mycomp is different and throws during copy.

`_Always_clear` is needed when:
1. the move assignment fails and cannot provide strong-guarantee on `_Other` side, or
2. the move assignment succeeds but leaves `_Other` in a broken state.
(The same goes for move-ctor's `extract`.)

Reply:
um, address checking is needed, otherwise self-move will cause the flat_set being emptied...